src/sage/matroids/chow_ring_ideal.py

```diff
@@ -0,0 +1,655 @@
+            Ag*Abcd, Ag*Ade, Ag*Adf, Ag*Aef, Aabf*Aace, Aabf*Aadg, Aabf*Abcd,
+            Aabf*Abeg, Aabf*Acfg, Aabf*Ade, Aabf*Adf, Aabf*Aef, Aace*Aadg,
+            Aace*Abcd, Aace*Abeg, Aace*Acfg, Aace*Ade, Aace*Adf, Aace*Aef,
+            Aadg*Abcd, Aadg*Abeg, Aadg*Acfg, Aadg*Ade, Aadg*Adf, Aadg*Aef,
+            Abcd*Abeg, Abcd*Acfg, Abcd*Ade, Abcd*Adf, Abcd*Aef, Abeg*Acfg,
+            Abeg*Ade, Abeg*Adf, Abeg*Aef, Acfg*Ade, Acfg*Adf, Acfg*Aef,
+            Ade*Adf, Ade*Aef, Adf*Aef,
+            -Ab + Ae - Aabf + Aace - Abcd + Ade + Aef,
+            -Ac + Ae - Abcd + Abeg - Acfg + Ade + Aef,
+            Ae - Ag + Aace - Aadg - Acfg + Ade + Aef,
+            -Ad + Ae + Aace - Aadg - Abcd + Abeg - Adf + Aef,
+            -Aa + Ae - Aabf - Aadg + Abeg + Ade + Aef,
+            Ae - Af - Aabf + Aace + Abeg - Acfg + Ade - Adf,
+            Ab - Ac + Aabf - Aace + Abeg - Acfg,
+            Ab - Ag + Aabf - Aadg + Abcd - Acfg,
+            Ab - Ad + Aabf - Aadg + Abeg - Ade - Adf,
+            -Aa + Ab - Aace - Aadg + Abcd + Abeg,
+            Ab - Af + Abcd + Abeg - Acfg - Adf - Aef,
+            Ac - Ag + Aace - Aadg + Abcd - Abeg,
+            Ac - Ad + Aace - Aadg + Acfg - Ade - Adf,
+            -Aa + Ac - Aabf - Aadg + Abcd + Acfg,
+            Ac - Af - Aabf + Aace + Abcd - Adf - Aef,
+            -Ad + Ag - Abcd + Abeg + Acfg - Ade - Adf,
+            -Aa + Ag - Aabf - Aace + Abeg + Acfg,
+            -Af + Ag - Aabf + Aadg + Abeg - Adf - Aef,
+            -Aa + Ad - Aabf - Aace + Abcd + Ade + Adf,
+            Ad - Af - Aabf + Aadg + Abcd - Acfg + Ade - Aef,
+            Aa - Af + Aace + Aadg - Acfg - Adf - Aef]
+        """
+        E = list(self._matroid.groundset())
+        flats = list(self._flats_generator.keys())
+        flats_containing = {x: [] for x in E}
+        for i,F in enumerate(flats):
+            for x in F:
+                flats_containing[x].append(i)
+        gens = poly_ring.gens()
+        Q = [gens[i] * gens[i+j+1] for i,F in enumerate(flats)
+            for j,G in enumerate(flats[i+1:]) if not (F < G or G < F)] #Quadratic Generators
+        L = [sum(gens[i] for i in flats_containing[x])
+            - sum(gens[i] for i in flats_containing[y])
+            for j,x in enumerate(E) for y in E[j+1:]] #Linear Generators
+        return Q + L
+
+    def _repr_(self):
+        r"""
+        EXAMPLES::
+
+            sage: ch = matroids.catalog.Fano().chow_ring(QQ, False)
+            sage: ch
+            Chow ring ideal of Fano: Binary matroid of rank 3 on 7 elements, type (3, 0)
+        """
+        return "Chow ring ideal of {}".format(self._matroid)
+
+    def groebner_basis(self, algorithm='constructed'): #can you reduce it? - consider every antichain of size 2, and chains?
+        r"""
+        Returns the Groebner basis of the Chow ring ideal.
+
+        EXAMPLES::
+
+            sage: from sage.matroids.basis_matroid import BasisMatroid
+
```

Comment:
```suggestion
        Return the Groebner basis of `self`.

        EXAMPLES::

            sage: from sage.matroids.basis_matroid import BasisMatroid
```
It would be better to use the standard Sage global namespace way to build the matroid rather than via an import.